### PR TITLE
feat(route53): verify alias `bucketName` matches record name

### PR DESCRIPTION
**Pull request by nmussy**
Enforces that fully qualified `A` record name matches the aliased `bucketName`

* add `IValidatableAliasRecordTarget` interface, validates alias against `CfnRecordSet`

Followup on #4248

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53-targets/test/bucket-website-target.test.ts [hunk 1]
```diff
@@ -93,3 +94,46 @@ test('throws if bucket website hosting is unavailable (cn-northwest-1)', () => {
...
 });
+
+
+test("throws if bucket name doesn't match full record domain name: new Bucket", () => {
+  // GIVEN
+  const app = new App();
+  const stack = new Stack(app, 'test', {env: {region: 'us-east-1'}});
+
+  // FIXME bucketName is unresolved?
+  const bucketWebsite = new s3.Bucket(stack, 'Bucket', { bucketName: ['different', bucketName].join('.') });
```
Comment (nmussy) [inline packages/@aws-cdk/aws-route53-targets/test/bucket-website-target.test.ts]: Any idea on how I could access `bucketName` there? The property is a token once I access it in the `validate` method

File: packages/@aws-cdk/aws-route53/lib/alias-record-target.ts [hunk 2]
```diff
@@ -11,6 +12,21 @@ export interface IAliasRecordTarget {
   bind(record: IRecordSet): AliasRecordTargetConfig;
 }
 
+/**
+ * Validatable alias record target class
+ */
+export interface IValidatableAliasRecordTarget extends IAliasRecordTarget {
```
Comment (nmussy) [inline packages/@aws-cdk/aws-route53/lib/alias-record-target.ts]: awslint seems to be complaining about the interface not being used, while I implement it in `bucket-website-target`:

```
@aws-cdk/aws-route53: error: [awslint:no-unused-type:@aws-cdk/aws-route53.IValidatableAliasRecordTarget] type or enum is not used by exported classes (declared at lib/alias-record-target.ts:18)
```

Comment (shivlaks) [reply packages/@aws-cdk/aws-route53/lib/alias-record-target.ts]: @nmussy did you need a hand with this one - I can pull it down to take a look at what awslint might be complaining about

**Comment by shivlaks** [comment]
closing this PR as it's stale to reflect that this task is available to pick up on the issue. PR can be reopened or a new one can be published in it's place.

**Pull request closed without merging**